File: app/src/main/java/com/sysu/edu/academic/CourseSelectionActivity.java
package com.sysu.edu.academic;

import android.os.Bundle;
import android.view.WindowManager;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.tabs.TabLayoutMediator;
import com.sysu.edu.R;
import com.sysu.edu.databinding.ActivityCourseSelectionBinding;
import com.sysu.edu.view.Pager2Adapter;

import java.util.List;
import java.util.stream.IntStream;

public class CourseSelectionActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityCourseSelectionBinding binding = ActivityCourseSelectionBinding.inflate(getLayoutInflater());
        binding.toolbar.setNavigationOnClickListener(_ -> supportFinishAfterTransition());
        setContentView(binding.getRoot());
        Pager2Adapter pager2Adapter = new Pager2Adapter(this);
        binding.pager2.setAdapter(pager2Adapter);
        new TabLayoutMediator(binding.tab, binding.pager2, (tab, position) -> tab.setText(List.of(R.string.course_selection, R.string.preview, R.string.course_selected).get(position))).attach();
        IntStream.range(0, 3).forEach(i -> pager2Adapter.add(CourseSelectionContainerFragment.newInstance(i)));
        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
    }
}

File: app/src/main/java/com/miyuyan/sysuer/academic/CourseSelectionActivity.kt
package com.miyuyan.sysuer.academic

import android.os.Bundle
import android.view.WindowManager
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import com.miyuyan.sysuer.BaseActivity
import com.miyuyan.sysuer.R
import com.miyuyan.sysuer.databinding.ActivityCourseSelectionBinding
import com.miyuyan.sysuer.view.Pager2Adapter

class CourseSelectionActivity : BaseActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val binding = ActivityCourseSelectionBinding.inflate(layoutInflater)
		binding.toolbar.setNavigationOnClickListener { supportFinishAfterTransition() }
		setContentView(binding.getRoot())
		val pager2Adapter = Pager2Adapter(this)
		binding.pager2.setAdapter(pager2Adapter)
		TabLayoutMediator(binding.tab, binding.pager2) { tab: TabLayout.Tab?, position: Int ->
			tab!!.setText(mutableListOf(R.string.course_selection, R.string.preview, R.string.course_selected)[position])
		}.attach()
		(0..<3).forEach { pager2Adapter.add(CourseSelectionContainerFragment.newInstance(it)) }
		window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN)
	}
}